retrospective-update/functions.py

```
@@ -0,0 +1,1036 @@
+        local_file_name = os.path.join(outputs_dir, vpu, os.path.basename(most_recent_qfinal))
+        if not os.path.exists(local_file_name):
+            s3.get(most_recent_qfinal, local_file_name)
+    return
+
+def verify_era5_data(runoff_dir: str, hourly_zarr: str, CL: CloudLog) -> None:
+    """
+    Verifies that the ERA5 data is compatible with the retrospective zarr
+    """
+    runoff_files = glob.glob(os.path.join(runoff_dir, '*.nc'))
+    if not runoff_files:
+        CL.ping('STOPPING', f"No-runoff-files-found")
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)